api/src/indigo_loaders.h

```diff
@@ -121,6 +136,24 @@ class IndigoRdfLoader : public IndigoObject
     AutoPtr<Scanner> _own_scanner;
 };
 
+
+class IndigoJSONMolecule : public IndigoObject
```

Comment:
Are you sure to inherit from `IndigoObject`? Maybe from `IndigoMolecule`, which already incapsulates `Molecule`?

Reply:
JSON Molecule is not a Molecule (e.g. JSON object can be SDF), so it is worth keep it separate

Reply:
BTW, why rejected the initial approach, when a loader was an operation on a molecule?
Introducing JSON Molecule, SDF Molecule, we bind domain objects to the format of storage. 
They are completely different objects and cannot be simply casted each other (I mean JSONModlecule to SDD Molecule or the like), we'll have to add special converters.